org.alloytools.alloy.application/src/main/java/se/src/alloy/cli/Args.java

```diff
@@ -0,0 +1,82 @@
+package se.src.alloy.cli;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class Args {
+  public final Config configFile;
+  private List<String> files;
+  private final Boolean watch;
+
+  private Args(Config configFile, List<String> files, Boolean watch) {
+    this.configFile = configFile;
+    this.files = files;
+    this.watch = watch;
+  }
+
+  public static Args parse(List<String> args) throws IOException {
```

Comment:
Maybe it would be worthwhile using an exiting CLI library instead of parsing our own arguments.  If not, custom argument parsing is fine as long as 
  - all the switches are declared in one place
  - there is a good help/usage message that is not hardcoded but constructed automatically from the declared switches